query-graphs-js/src/querygraph.ts

```diff
@@ -1092,8 +1105,7 @@ class GraphBuilderFromSchema extends GraphBuilder {
     private readonly supergraph?: { apiSchema: Schema, isFed1: boolean },
   ) {
     super();
-    this.isFederatedSubgraph = isFederationSubgraphSchema(schema);
-    assert(!this.isFederatedSubgraph || supergraph, `Missing supergraph schema for building the federated subgraph graph`);
+    this.isFederatedSubgraph = !!supergraph && isFederationSubgraphSchema(schema);
```

Comment:
This one change is the only exception to "this just adds `provideId`. The reason for this is that the assertion was getting in the way of creating a standalone query graph for a simple subgraph (outside of a full federated query graph that is), which is never done by the code in practice, but can still be nice to do when debugging/learning about query graphs (basically to be able to visualise what is created for a single subgraph in isolation). Again, shouldn't change anything for existing code.